File: app/controllers/select_category_controller.rb
class SelectCategoryController < UITableViewController

  attr_accessor :scratch_period

  #### table view delegate methods ####

  def tableView(tableView, numberOfRowsInSection: section)
    Category.count
  end

  def tableView(tableView, cellForRowAtIndexPath: indexPath)
    cell = tableView.dequeueReusableCellWithIdentifier("SelectCategoryCell")
    cell ||= UITableViewCell.alloc.initWithStyle(UITableViewCellStyleDefault, reuseIdentifier: "SelectCategoryCell")
    category = Category.where(:index).eq(indexPath.row).first
    cell.nameLabel.text = category.name
    cell.colorMark.color = category.color
    if @scratch_period[:category]
      cell.accessoryType = UITableViewCellAccessoryCheckmark if category == @scratch_period[:category]
    end
    cell
  end

  def tableView(tableView, didSelectRowAtIndexPath: indexPath)
    @scratch_period[:category] = Category.where(:index).eq(indexPath.row).first
    
    self.navigationController.popViewControllerAnimated(true)
  end

end

File: Forking Pasta/SelectCategoryController.swift
import UIKit
import KingPastaKit

class SelectTimeZoneController: UITableViewController {
    
    var delegate: PeriodController!
    var periodTimeZone: TimeZone?
    
    // MARK: UITableViewDataSource
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return TimeZone.count
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("SelectTimeZoneCell") as? SelectTimeZoneCell ??
            SelectTimeZoneCell(style: .Default, reuseIdentifier: "SelectTimeZoneCell")
        let timeZone = TimeZone.forIndex(indexPath.row)
        cell.nameLabel.text = timeZone.name
        cell.colorMark.color = timeZone.color
        if let periodTimeZone = periodTimeZone where timeZone == periodTimeZone {
            cell.accessoryType = .Checkmark
        }
        
        return cell
    }
    
    // MARK: UITableViewDelegate
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        delegate.periodTimeZone = TimeZone.forIndex(indexPath.row)
        navigationController?.popViewControllerAnimated(true)
    }
}